open_humans/templatetags/utilities.py

```diff
@@ -368,16 +368,29 @@ def template_bool(item):
 
 
 @register.simple_tag(takes_context=True)
-def get_next_url(context):
-    next = context['request'].GET.get('next', None)
-    if next:
-        return quote_plus(next)
+def get_next_url(context, include_next=False, social=False):
+    """
+    Get the url to assign to ?next=.
```

Comment:
There's some complicated logic going on here! Seems like a lot of scenarios? I expect it does all make sense, but maybe you can walk me through it when we chat... :)